analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/LooseFilePermissions.cs

```diff
@@ -0,0 +1,152 @@
+    [Rule(DiagnosticId)]
+    public sealed class LooseFilePermissions : HotspotDiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "S2612";
+        private const string MessageFormat = "Make sure this permission is safe.";
+        private const string Everyone = "Everyone";
+
+        private static readonly HashSet<string> WeakFileAccessPermissions = new HashSet<string>
+        {
+            "AllPermissions",
+            "DefaultPermissions",
+            "OtherExecute",
+            "OtherWrite",
+            "OtherRead",
+            "OtherReadWriteExecute"
+        };
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager).WithNotConfigurable();
+
+        public LooseFilePermissions() : this(AnalyzerConfiguration.Hotspot) { }
+
+        internal LooseFilePermissions(IAnalyzerConfiguration configuration) : base(configuration)
+        {
+        }
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterCompilationStartAction(c =>
+            {
+                if (!IsEnabled(c.Options))
+                {
+                    return;
+                }
+
+                c.RegisterSyntaxNodeActionInNonGenerated(VisitInvocations, SyntaxKind.InvocationExpression);
+
+                c.RegisterSyntaxNodeActionInNonGenerated(VisitAssignments, SyntaxKind.IdentifierName);
+            });
+
+        private static void VisitAssignments(SyntaxNodeAnalysisContext context)
+        {
+            var identifier = (IdentifierNameSyntax)context.Node;
+            if (IsFileAccessPermissions(identifier, context.SemanticModel) && !identifier.IsPartOfBinaryNegationOrCondition())
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, identifier.GetLocation()));
+            }
+        }
+
+        private static bool IsFileAccessPermissions(SimpleNameSyntax identifierNameSyntax, SemanticModel semanticModel) =>
+            WeakFileAccessPermissions.Contains(identifierNameSyntax.Identifier.Text)
+            && identifierNameSyntax.IsKnownType(KnownType.Mono_Unix_FileAccessPermissions, semanticModel);
+
+        private static void VisitInvocations(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = (InvocationExpressionSyntax)context.Node;
+            if (!invocation.IsMemberAccessOnKnownType("SetAccessRule", KnownType.System_Security_AccessControl_FileSystemSecurity, context.SemanticModel) &&
+                !invocation.IsMemberAccessOnKnownType("AddAccessRule", KnownType.System_Security_AccessControl_FileSystemSecurity, context.SemanticModel))
+            {
+                return;
+            }
+
+            if (GetObjectCreation(invocation, context.SemanticModel) is { })
```

Comment:
Maybe the the two ifs could be merged if( (SetAccessRule || AddAccessRule) && GetObjectCreation)